### PR TITLE
Allow buffer-pack-in-one also for `Mesh::Initialize`

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

Existing solution applied to new location.

Before (`Mesh::Initialize` takes 1.63s)

Screenshot from 2021-04-27 21-02-47

After ((`Mesh::Initialize` takes 385ms)

Screenshot from 2021-04-27 21-03-00

## PR Checklist

- [x] Code passes cpplint
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: src/mesh/mesh.cpp [hunk 1]
```diff
@@ -1042,15 +1044,34 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
...
+        if (cell_centered_bvars::ReceiveBoundaryBuffers(md) != TaskStatus::complete) {
+          all_received = false;
+        }
+      }
+    } while (!all_received);
+
+    // unpack FillGhost variables
+    for (int i = 0; i < num_partitions; i++) {
+      auto &md = mesh_data.GetOrAdd("base", i);
+      cell_centered_bvars::SetBoundaries(md);
```
Comment (Yurlungur) [inline src/mesh/mesh.cpp]: This is such a small changeset. Can we cage this behind an ifdef or runtime parameter? That way codes using sparse variables will still run and other codes can benefit.

**Review by Yurlungur — APPROVED** [review]
Nice to see the work for `pack-in-one` bearing fruit in this small changelog. See my one comment below, otherwise I'm happy.

**Comment by pgrete** [comment]
Ready for review again. Now there's an option, which is used in the short test (i.e., explicitly enabled) and then we test the default (off) in the long test

**Review by Yurlungur — APPROVED** [review]

**Review by jlippuner — APPROVED** [review]
Wonderful, thanks!

**Pull request merged**